overlord/servicestate/servicestate.go

```diff
@@ -61,9 +62,9 @@ func computeExplicitServices(appInfos []*snap.AppInfo, names []string) map[strin
 
 	for _, app := range appInfos {
 		snapName := app.Snap.InstanceName()
-		appName := app.String()
-		if requested[appName] {
-			explicitServices[snapName] = append(explicitServices[snapName], app.ServiceName())
+		// app.String() gives "snapname.appname"
+		if requested[app.String()] {
+			explicitServices[snapName] = append(explicitServices[snapName], app.Name)
```

Comment:
this change is not backward compatible though?  I think the current values are already used in stable, no? this is a potential problem with reverts

Reply:
oh, I though it was only in edge. Oh well, yes, it's not backwards compatible.

Reply:
I might be wrong which would make our life easier because we haven't done a release in long time

Reply:
sorry, I was confused ExplicitServices is not in 2.51:

https://github.com/snapcore/snapd/blob/release/2.51/overlord/servicestate/service_control.go#L42

so this in principle is ok. But I don't remember why we went with the full service names, we need to be a bit careful